app/models/project.rb

```diff
@@ -170,10 +170,11 @@ def communication_emails
     users_with_project_roles(%w(owner communications)).pluck(:email)
   end
 
-  def notify_on_subject_set_completion?
-    # default behaviour will be to opt in to subject set completion notifications
-    # but we can by changing this below on the fetch fallback.
-    notify_value = configuration.fetch('notify_on_subject_set_completion', 'false')
-    notify_value.to_s.casecmp('true').zero?
+  def run_subject_set_completion_events?
+    # short term use ENV vars to 'allowlist' projects into this functionality - so we control it
+    # longer term perhaps switch to opt in behaviours via project configuration setting?
+    # or should this be an admin only flag for our control??
+    allow_list_project_ids = ENV.fetch('SUBJECT_SET_COMPLETION_EVENT_PROJECT_IDS', '').split(',').map(&:to_i)
```

Comment:
this could easily be moved to an explicit, admin-only attribute on the project resource that we control via the admin page. TBC

Reply:
As it's currently setup through ENV vars - your understanding is correct. The more i think about this i think we should move it to an admin only `project.run_subject_set_completion_events` boolean attribute (default is false)

Moving it to an admin attribute will allow zoo team members to opt in / out of this functionality via the admin pages / API  and avoid having to ask a dev team member to update some env vars and deploy.

Also longer term it's easy to switch that attribute to non-admin and allow project teams to opt into this behaviour (via lab interface or the API).

I think i should do that before merging... Thoughts?

Reply:
I'll merge this for now and follow up with a PR to move this to an admin only project attribute.

I'm mostly concerned with the organizational time debt where a zoo team member has to rope a developer in and merge a PR and deploy etc just to add this feature for a project.